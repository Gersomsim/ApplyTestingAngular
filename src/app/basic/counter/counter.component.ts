import { Component } from '@angular/core';

@Component({
  selector: 'app-counter',
  templateUrl: './counter.component.html',
})
export class CounterComponent {
  counter = 10;

  increaseBy(value: number){
    this.counter += value
    // console.log('new value: ', this.counter)
  }

}
