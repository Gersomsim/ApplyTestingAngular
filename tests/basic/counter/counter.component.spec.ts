import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../src/app/basic/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compile: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CounterComponent]
    });
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compile = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('Debe de hacer mach con el snapshot', () => {
    expect(compile).toMatchSnapshot()
  })

  test('increaseBy debe de incrementar basado en el argumento (5)', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(15)
  });

  test('Hacer click en los botones debe de incrementar y decrementar en 1', () => {
    const buttons = compile.querySelectorAll('button');
    buttons[0].click();
    expect(component.counter).toBe(11);
    buttons[1].click();
    buttons[1].click();
    expect(component.counter).toBe(9);
  })

  test('Cambiar el counter debe de actualizar el tag h1', () => {

    component.increaseBy(10);
    fixture.detectChanges()

    const h1 = compile.querySelector('h1');

    expect(h1?.textContent).toContain('20');
  });
});
