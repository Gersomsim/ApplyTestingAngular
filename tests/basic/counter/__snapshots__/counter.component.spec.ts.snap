// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent Debe de hacer mach con el snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 10
  </h1>
  <hr />
  <button
    class="btn btn-primary mx-1"
  >
    +1
  </button>
  <button
    class="btn btn-primary mx-1"
  >
    -1
  </button>
</div>
`;
